go-controller/pkg/factory/factory_test.go

```diff
@@ -394,57 +394,126 @@ var _ = Describe("Watch Factory Operations", func() {
 					defer GinkgoRecover()
 					Expect(len(objs)).To(Equal(1))
 					return nil
-				})
+				}, wf.GetHandlerPriority(objType))
 			Expect(h).NotTo(BeNil())
 			Expect(err).NotTo(HaveOccurred())
+			Expect(h.priority).To(Equal(priority))
+			wf.removeHandler(objType, h)
+		}
+
+		testExistingFilteredHandler := func(objType reflect.Type, realObj reflect.Type, namespace string, sel labels.Selector, priority uint32) {
```

Comment:
maybe you could add a test where you increment some shared variable with handlers like:
1. y = 0
2. add handler with priority 1, assert y=0, increment y
3. add handler with priority 2, assert y=1, multiply y by 10
4. add handler with priority 3, assert y=10, subtract 2
5. assert variable is now 8

some basic math where order of operations matter, so you can be sure the handlers executed in order

This would make sure the handlers were incremented in the right order.

Reply:
ah yeah! this is something we should add for sure!